### ISSUE TITLE
Connecting to Heroku PostgreSQL

### ISSUE THREAD

**Issue by mohamead** [strapi/strapi#2522]
**Informations**
- **Node.js version**:  v10.14.2
- **NPM version**: 6.4.1
- **Strapi version**:  3.0.0-alpha.17
- **Database**: PostgreSQL
- **Operating system**: Windows 10
- **(Optional) Link to your Project**: 

**What is the current behavior?**
Can not connect to heroku PostgreSQL

**Steps to reproduce the problem**
Running the following command in the terminal:

`strapi new Test --dbclient=postgres --dbhost=herokuHost --dbport=5432 --dbname=herokuDbName --dbusername=herokuUsername --dbpassword=herokuPassword --dbssl=false --dbauth=herokuDbName
`

Error returned: 

`Start creating your Strapi application. It might take a minute, please take a coffee`
`Lets configurate the connection to your database:`
`Database determined by CLI args: postgres`
`Testing database connection...`
`Database connection has failed! Make sure your "herokuDbName" database exist.`

**What is the expected behavior?**

Connection is created successfully 

**Suggested solutions**

...

**Comment by Skeyelab** [comment]
@lauriejim Have you created your `herokuDbName` database on your server?

**Comment by derrickmehaffy** [comment]
@Skeyelab I'm guessing you ment to tag @mohamead

**Comment by Skeyelab** [comment]
@mohamead - edit: i tried doing what you are doing.  I am getting the same result.  Heroku creates the DB for us.  I can also successfully connect from a local client.

**Comment by lauriejim** [comment]
Hello! Yap I have the same issue.
Before fixing it we have to modify the way we handle the connectivity script in development mode to make it easier to debug.

**Comment by frey1esm** [comment]
I am having the same issue with MySql.  My database exists, the user exists, password is correct, and I can connect to the database from the command line. But 'strap new my_project` throws the error `Database connection has failed! Make sure your "db_myapp" database exist.`

**Comment by yao-dev** [comment]
I was correct the bug with adding this code at line 213 of the file /usr/local/lib/node_modules/strapi/node_modules/strapi-generate-new/lib/before.js (where strapi is installed globally on your machine) : 
```
scope.database.settings.ssl = scope.database.settings.client === 'postgres' ? true : false;
```

Now I'm able to generate a project with a remote postgres database

**Comment by mohamead** [comment]
@yao-dev I tried the fix that you suggested and it works, but just so that I fully understand. The line that you added is to force enabling SSL when connecting to Postgres? Which is a precondition for connection to Heroku Postgres as said her -> https://devcenter.heroku.com/articles/connecting-to-heroku-postgres-databases-from-outside-of-heroku#ssl

But isn't this something that should be done when adding the option --dbssl?

When changing the code at line 212 in the same file

from
`scope.database.options.ssl = _.toString(answers.ssl) === 'true';`
to 
`scope.database.settings.ssl = _.toString(answers.ssl) === 'true';`

also fixed this issue, and retaining the indent feature. That is when connection to remote Postgres (e.g Heroku) one must use the option --dbssl=true, but this is not required when connecting locally.

 @lauriejim what do you think?

**Comment by yao-dev** [comment]
@mohamead When you choose postgres as database maybe it's better for Strapi to ask if you want enable ssl. Because in before.js a file is generated you can log "scope.tmpPath" to see the content of the file, it's look like : 

``` 
'use strict';

// Public node modules
const inquirer = require('inquirer');
const rimraf = require('rimraf');

module.exports = (scope, success, error) => {
  // eslint-disable-next-line import/no-unresolved
  const knex = require('knex')({
    client: scope.client.module,
    connection: Object.assign({}, scope.database.settings, {
      user: scope.database.settings.username
    })
  });

  knex.raw('select 1+1 as result').then(() => {
    knex.raw(scope.client.database === 'postgres' ? "SELECT tablename FROM pg_tables WHERE schemaname='public'" : 'SELECT * FROM information_schema.tables').then((tables) => {
      knex.destroy();

      const next = () => {
        rimraf(scope.tmpPath, (err) => {
          if (err) {
            console.log(`Error removing connection test folder: ${scope.tmpPath}`);
          }
          success();
        });
      };

      if (tables.rows && tables.rows.length !== 0) {
        console.log('🤔 It seems that your database is not empty. Be aware that Strapi is going to automatically creates tables & columns, and might update columns which can corrupt data or cause data loss.');

        inquirer.prompt([{
          type: 'confirm',
          name: 'confirm',
          message: `Are you sure you want to continue with the ${scope.database.settings.database} database:`,
        }])
          .then(({ confirm }) => {
            if (confirm) {
              next();
            } else {
              error();
            }
          });
      } else {
        next();
      }
    });
  })
    .catch((err) => {
      if (err.sql) {
        console.log('⚠️ Server connection has failed! Make sure your database server is running.');
      } else {
        console.log(`⚠️ Database connection has failed! Make sure your "${scope.database.settings.database}" database exist.`);
      }
      error();
    });
};

```

In before.js scope.database.options.ssl is always false because strapi don't ask in the prompt and in the generated file the object used to create database connection with knex is scope.database.settings not scope.database.options and the property "ssl" doesn't exist in scope.database.settings.

Because scope.database.settings.ssl = true is missing, the query in the generated file will failed : 

```
knex.raw(scope.client.database === 'postgres' ? "SELECT tablename FROM pg_tables WHERE schemaname='public'" : 'SELECT * FROM information_schema.tables')
```
I think it's to better to ask in the prompt if you want enable SSL

**Comment by mohamead** [comment]
@yao-dev updating the prompt sounds like a good idea, but that is something that I can not test as the prompt will freeze on windows (https://github.com/strapi/strapi/issues/1281). 

As described in the ticket above that is the OS that I' am currently using unfortunately. Hence the point about the option `--dbssl=true`. So to summaries running the command defined under "Steps to reproduce ...", but with the option `--dbssl=true` will solve this issue. Given the line change mentioned before..

**Comment by Skeyelab** [comment]
@mohamead, is this something that has been released in 3.0.0-alpha.18?

**Comment by derrickmehaffy** [comment]
@Skeyelab his PR was not merged so no it wasn't resolved yet.
